plugins/default/endpointReachability.ts

```diff
@@ -0,0 +1,34 @@
+import { PluginHandler } from '../types';
+import { getText } from '../utils';
+
+const endpointReachability = async (url: string): Promise<boolean> => {
+  if (!url) return false;
+  try {
+    // Making a HEAD request as its lightweight (gets headers and doesn't wait for response)
+    const response = await fetch(url, { method: 'HEAD' });
+    return response.ok;
+  } catch (error) {
+    // On all other cases like 403, 404, 500
+    return false;
+  }
+};
+
+export const handler: PluginHandler = async (
+  context,
+  _parameters,
+  eventType,
+  _options
+) => {
+  let error = null;
+  let verdict = false;
+  let data = null;
+
+  try {
+    let text = getText(context, eventType);
```

Comment:
My suggestion here is use regex to get all the URLs available from text and validate them using `URL` or some other way. 

With this we can have config weather to run plugin for all the endpoints or first endpoint found within the text.

What do you think? @5war00p

Reply:
Your thought makes sense, but this PR is a plugin PR which I think doesn't make sense to use it as utility function. I think we should focus on keep this as individual plugin. keeping that I prefer we should go for regex match. 

As an enhancement I think we shouldn't just blindly hit the URL, we should ask user which type of method they're thinking of ex: `GET` and `POST` and let plugin do `OPTIONS` request and make sure the endpoint(s) support the user requested methods.

let me know if this makes sense to you.

Reply:
I think if thats the case we should make it as a new separate plugin.